yt_dlp/YoutubeDL.py

```diff
@@ -4080,6 +4083,9 @@ def _write_thumbnails(self, label, info_dict, filename, thumb_filename_base=None
                 return ret
         multiple = write_all and len(thumbnails) > 1
 
+        if not self._ensure_dir_exists(filename):
+            return None
+
```

Comment:
This should be moved below the next block of code (just before the `for` loop), so that we don't make directories when thumb writing is skipped